src/frequenz/dispatch/_bg_service.py

```diff
@@ -36,13 +36,24 @@ class QueueItem:
         """A queue item for the scheduled events."""
 
         time: datetime
+        stop_event: bool
```

Comment:
```suggestion
        is_stop_event: bool
```

Otherwise it sounds like an action? Or even maybe an `event` enum with `START`/`STOP` as it would be more extensible for other potential events? But probably that's overkill at this stage.

BTW, I still don't understand what this is for, it even seems to be unused!

Reply:
Given two events at the same time:
* Dispatch 12 - STOP
* Dispatch 13 - START

As the list of events is ordered based on (event time, id),
we would in this case first run dispatch 12 as the time matches and the id smaller.

What we want, however is to run always the START event if the time is equal. Because then the filters looking at the events had a chance to observe START before STOP and thus can filter out the STOP when desired.

> it even seems to be unused!

We don't need to directly reference it, the ordering is done by the dataclass

Reply:
Oh, this is really sneaky! So you are relying on the order between `True` and `False`? You naughty naughty boy...

Then I would definitely clarify this in a comment and probably even make it a `priority: int` (or a `IntEnum`) to make it even more clear that this is in the item only to affect the order in the queue.

Reply:
Sneaky? I all that elegant.